tools/format-file.py

```diff
@@ -0,0 +1,226 @@
+			lineComments.append(lineComment)
+			# read comma-separated cells and trim them
+			cellsCleaned = []
+			for (c, cell) in enumerate(cellsRaw):
+				cell = cellsRaw[c];
+				cell = re.sub('/\*.*\*\/', r'', cell)
+				cellsCleaned.append(cell.strip())
+			cells.append(cellsCleaned)
+		lineComments = lineComments[1:]
+		colWidth = max(len(cell) for cell in cells)
+
+		# for each column append padding to cells
+		for c in range(0, colWidth):
+			# find maxlen of the column -> column width
+			maxlen = 0
+			for (r, row) in enumerate(cells):
+				if(len(row) > c and len(row[c]) > maxlen):
+					maxlen = len(row[c])
+			# replace cells of column with padded string
+			for (r, row) in enumerate(cells):
+				if(len(row) > c):
+					cell = row[c]
+					if(cell == ""): continue
+					if(len(row) == (c + 1)):
+						cell += ' ' # no comma after last cell in row
+					else:
+						cell += ','
+					fill = spaces[0:(maxlen - len(row[c]))]
+					cell += fill
+					cells[r][c] = cell
+
+		# put cells back together to a table
+		tableInner = ""
+		for (r, row) in enumerate(cells):
+			tableInner += '{ ' + ' '.join([item for item in row]) + '}'
+			lineComment = ''
+			if(len(lineComments) > r):
+				# print('lc: ' + str(r) + ' --> ' + lineComments[r])
+				lineComment = lineComments[r]
+				# reindent multiline comments
+				lineComment = re.sub('\n\s*', r'\n' + indent + '        ', lineComment)
+			if(len(cells) > (r + 1)):
+				tableInner += ',' + lineComment + '\n' + indent + '        '
+
+		# write table
+		tableStart = '{\n' + indent + '        '
+		tableEnd = '\n' + indent + '};'
+		table = tableStart +  tableInner + tableEnd
+		table = re.sub('{\s*}', r'{}', table)
+		content = content.replace(occurence, table)
+	return content
+
+if __name__ == '__main__':
+	opts, args = getopt.getopt(sys.argv[1:], 'c:hd', ['clang-format=', 'help', 'dry'])
+	dry = False
+	for opt, arg in opts:
+		if opt in ('--help'):
+			printHelp()
+		if opt in ('--clang-format'):
+			clangFormatCmd = arg
+		if opt in ('--dry'):
+			dry = True
+	files = args
+
+	if(len(files) <= 0):
+		print('error: no file given')
+		printHelp()
+
+	if (shutil.which(clangFormatCmd) is None):
+		print('error: "' + clangFormatCmd + '" not executable')
+		printHelp()
+
+	versionString = str(check_output(split(clangFormatCmd + ' -version')))
+	if ("version 6.0" not in versionString):
+		print('error: "' + clangFormatCmd + '" not in correct version, expecting version 6.0.x')
+		print('got version: "' + versionString)
+		printHelp()
+
+	print("FORMATTING: ")
+	for file in files:
+		print(file.ljust(40), end="", flush=True)
+		if(not os.path.isfile(file)):
+			print(' [Skipping non existing file]')
+			continue
+		if(file.find('MurmurHash2') > 0):
+			print(' [Skipping MurmurHash2 file]')
+			continue
+		fileTmp = file + '.tmp'
+		shutil.copyfile(file, fileTmp)
+		run_clang_format(fileTmp)
+		run_post_clang_format(fileTmp)
+		if(filecmp.cmp(file, fileTmp)):
```

Comment:
It'd be nicer to have the temporary file renamed over the target file, not copied. Then the operation is atomic.

Reply:
i will change that...

Reply:
Done